staging/src/k8s.io/apiserver/pkg/server/options/recommended.go

```diff
@@ -122,14 +122,23 @@ func (o *RecommendedOptions) ApplyTo(config *server.RecommendedConfig) error {
 	if err := o.CoreAPI.ApplyTo(config); err != nil {
 		return err
 	}
+	if err := o.EgressSelector.ApplyTo(&config.Config); err != nil {
+		return err
+	}
+	if feature.DefaultFeatureGate.Enabled(features.APIServerTracing) {
+		// Apply tracing config right after CoreAPI and EgressSelector so that the rest config is
+		// updated with a TracerProvider before anyone tries to use it.
+		if wrapper, err := o.Traces.ApplyTo(config.Config.EgressSelector, &config.Config); err != nil {
+			return err
+		} else if config.TracerProvider != nil {
+			config.ClientConfig.Wrap(wrapper)
+		}
+	}
```

Comment:
is the ordering significant or did you just group these for readability?

Reply:
The tracing `ApplyTo` function relies on the EgressSelector already being configured, and should come immediately after the CoreAPI ApplyTo so that any users of the client use our wrapper, and get traces setup.

Reply:
`clientgoExternalClient, err := clientgoclientset.NewForConfig(kubeconfig)` is constructed/used inside CoreAPIOptions#ApplyTo ... that won't have tracing

Reply:
I moved wrapping the client to inside CoreAPI.  Conveniently, it now captures both the external client, and the one embedded in config.